File: java/3174.java
class Solution {
    public String clearDigits(String s) {
        char stack[] = new char[s.length()];
        int i = 0;
        for (char c : s.toCharArray())
            if (Character.isDigit(c))
                --i;
            else
                stack[i++] = c;
        if (i <= 0)
            return "";
        return new String(stack, 0, i);
    }
}

File: c/3174.c
char* clearDigits(char* s) {
    int i = 0, n = strlen(s);
    for (int j = 0; j < n; j++)
        if (s[j] <= '9')
            --i;
        else
            s[i++] = s[j];
    if (i <= 0)
        s[0] = '\0';
    else
        s[i] = '\0';
    return s;
}
